### PR TITLE
Improved code design for GUI, prompts. Textareas replaced for a simple code editor.

### PR DESCRIPTION
**Pull request by ghost**
Implemented a code editor to replace the textareas. This provides syntax highlighting for easier writing. Still implementing it, functionally-wise is ready, GUI-wise it's still a little unstable, resizing and overflow are the issues.

The prompt for GPT has been improved. This helps GPT understand what it needs to do more precisely.

Various improvements all across the GUI to make it responsive.

JOB IN PROGRESS.

### CODE REVIEW

**Comment by ghost** [comment]
image
New look for the code in the GUI.

Resolved all the issues for the resizing and overflow and refactored some innecesary code and some that could be improved.

File: src/app/promptTemplates.ts [hunk 1]
```diff
@@ -1,20 +1,33 @@
-import { ShotsTypes, userInputsArr } from '@/store/inputStore';
+import { ShotsTypes } from '@/store/inputStore';
 
-export function insertExamples(examples:ShotsTypes[]){
-    const msjs:any[] = [];
-    msjs.push({"role": "System", "content":"You are a code generator machine"});
+const masterPrompt = (input:string):string => {
```
Comment (oscarfgq) [inline src/app/promptTemplates.ts:3]: Wouldn't it be better with template literals?

**Review by oscarfgq — APPROVED** [review]

File: src/app/promptTemplates.ts (hunk 1 above)
Comment (ghost) [reply src/app/promptTemplates.ts:3]: It is far better, thank you

File: src/app/components/Examples.tsx [hunk 2]
```diff
@@ -1,35 +1,36 @@
...
+export function Examples({ send }: ExampleTypes){
 
     const [deleteUserInputArr,           
            examples] = userInputsArr((state)=> [state.deleteUserInputArr,
                                                 state.examples]);
-    //3 shots minimun must be added to send a response request to GPT                                            
-    const disabled = examples.length >= 3 ? false : true;
+    const [userInputs] = userInputStore((state)=>[state.userInputs]);
+    //1 shot minimun must be added to send a response request to GPT                                            
+    const disabled = examples.length >= 1 && userInputs !== '' ? false : true;
```
Comment (AntonioGGovea) [inline src/app/components/Examples.tsx]: this can be simplified, think carefully

File: src/app/components/Inputs.tsx [hunk 3]
```diff
@@ -17,24 +18,26 @@ function Inputs({ data }:InputsType){
                                                          state.userResults,
                                                          state.updateUserInput,
                                                          state.updateUserResults]);   
-                                                         
-    const clickedAdd = () => { 
-        if((userInput.trim() !== '' && userResults.trim() !== '') ||
-           (userInput.trim() !== '' && examples.length >= 3)){
 
+    const readyToSend = examples.length > 0 && examples[examples.length-1].result === '' ? true : false;
```
Comment (AntonioGGovea) [inline src/app/components/Inputs.tsx]: same here

**Review by AntonioGGovea — CHANGES_REQUESTED** [review]

**Review by AntonioGGovea — APPROVED** [review]

**Pull request merged**